### DISCUSSION TITLE
.NET 6 End-of-Life, .NET 8 Runtime Upgrade

### DISCUSSION THREAD

**Discussion by MapleWheels** [FakeFishGames/Barotrauma#15306] [Code questions and modding support]
Good day Undertow,

If you are unaware, .NET 6 reached it's end of life in [November, 2024](https://devblogs.microsoft.com/dotnet/dotnet-6-end-of-support/#:~:text=Closing-,.,technical%20support%20will%20be%20offered). Are you considering moving to .NET 8 (LTS) anytime soon, considering security, performance and platform bug fixes? 

We're wondering at [LuaCsForBarotrauma](https://github.com/evilfactory/LuaCsForBarotrauma/tree/refactor/luacs-services-model) because our choice of architecture for some things in the refactored APIs will depend on some newer language features, if they would be available.

Thanks and have a good day.

MapleWheels/PerfidiousLeaf.

**Comment by Regalis11** [comment]
Thanks for the heads-up, I at least wasn't aware of this. :)

We're going to look into what upgrading to .NET 8 would entail and see how we're going to handle this. I unfortunately can't give any ETA on when I can get back to you on this, but I'm hoping we could do some decisions on this early next year.

**Comment by Regalis11** [comment]
We've now investigated this a bit and decided to upgrade to .NET 8. It will be a while before the upgrade is included in the public version of the game though: our plan is to include it in the update that introduces the code modding support we're currently working on (not sure if you're aware we are - this hasn't been publicly announced yet, but you may have seen some commit messages about this on Discord). The public release of that feature is probably going to be somewhere closer to the end of the year, or possible early 2026.

**Reply by MapleWheels** [reply]
Sounds great, if you need a hand or feedback on it, my DMs are open here or on Discord.

Edit: Yes, me and evil saw the commits mentioning it.
